pkg/bgpv1/manager/reconcilerv2/pod_ip_pool_test.go

```diff
@@ -0,0 +1,562 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package reconcilerv2
+
+import (
+	"context"
+	"net/netip"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+
+	"github.com/cilium/cilium/pkg/bgpv1/manager/instance"
+	"github.com/cilium/cilium/pkg/bgpv1/manager/store"
+	"github.com/cilium/cilium/pkg/bgpv1/types"
+	ipamtypes "github.com/cilium/cilium/pkg/ipam/types"
+	v2api "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	"github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	slimv1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/meta/v1"
+)
+
+var (
+	podIPPoolTestLogger = logrus.WithField("unit_test", "reconcilerv2_podippool")
+)
+
+var (
+	redPoolName       = "red-pool"
+	redPoolNamespace  = "default"
+	redLabelSelector  = slimv1.LabelSelector{MatchLabels: map[string]string{"pool": "red"}}
+	redNameNSSelector = slimv1.LabelSelector{MatchLabels: map[string]string{
+		podIPPoolNameLabel:      redPoolName,
+		podIPPoolNamespaceLabel: redPoolNamespace,
+	}}
+	redPool = &v2alpha1.CiliumPodIPPool{
+		ObjectMeta: metaV1.ObjectMeta{
+			Name:      redPoolName,
+			Namespace: redPoolNamespace,
+			Labels:    redLabelSelector.MatchLabels,
+		},
+		Spec: v2alpha1.IPPoolSpec{
+			IPv4: &v2alpha1.IPv4PoolSpec{
+				CIDRs:    []v2alpha1.PoolCIDR{"10.0.0.0/16"},
+				MaskSize: 24,
+			},
+			IPv6: &v2alpha1.IPv6PoolSpec{
+				CIDRs:    []v2alpha1.PoolCIDR{"2001:0:0:1234::/64"},
```

Comment:
Let's use ULA or documenting address for IPv6.

Reply:
good point, changes all tests to have `2001:db8::`  prefix.